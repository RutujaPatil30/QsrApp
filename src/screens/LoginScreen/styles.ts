import {StyleSheet} from 'react-native';
import theme from '../../utils/themes';
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: wp('100%'),
    alignSelf: 'center',
    backgroundColor: theme.colors.white,
  },
  subContainer: {
    width: wp('90%'),
    alignSelf: 'center',
  },
  imageContainer: {
    alignItems: 'center',
  },
  image: {
    width: wp('30%'),
    height: wp('15%'),
    resizeMode:'contain'
  },
  accoutExistStyle: {
    color: theme.colors.black,
    fontSize: theme.fonts.authLabel
  },
  accoutExistBtnStyle: {
    color: theme.colors.primary,
    fontSize: theme.fonts.authLabel
  },
  accountExistViewStyle: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    marginBottom: theme.fonts.authLabel,
  },
  errorText: {
    fontSize: theme.fonts.valueLabel,
    color: theme.colors.primary,
  },
  errorInput: {
    borderColor: theme.colors.primary,
  },
  button: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginVertical: wp('2.5%'),
  },
  viewHeaderStyle: {
    paddingVertical: wp('2.5%'),
    alignItems: 'center',
  },
  textHeaderStyle: {
    fontSize: theme.fonts.buttonText,
    color: theme.colors.black,
    fontWeight: '600',
  },
});
