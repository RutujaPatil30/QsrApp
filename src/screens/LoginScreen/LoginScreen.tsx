import { Formik } from 'formik';
import React, { useRef } from 'react';
import {
    Alert,
    Image,
    KeyboardAvoidingView,
    SafeAreaView,
    ScrollView,
    StatusBar,
    Text,
    TouchableOpacity,
    View
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import CustomButton from '../../components/CustomButton/CustomButton';
import CustomTextInput from '../../components/CustomTextInput/CustomTextInput';
import { login_request } from '../../redux/action';
import baselocalization from '../../utils/baselocalization';
import { images } from '../../utils/images';
import theme from '../../utils/themes';
import { ValidationSchema } from '../../utils/validations';
import { styles } from './styles';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';


const LoginScreen = (props: any) => {
    const { navigation } = props;
    const dispatch = useDispatch();
    const users = useSelector((state: any) => state.reducer.users);
    const initialValues = {
        email: '',
        password: '',
    };
    const formikRef = useRef();

    const onSubmit = (values: any) => {
        const user = users.find((user: any) => user.email === values.email);
        const userExist = users.some((user: any) => user.email === values.email);
        if (!userExist) {
            Alert.alert(
                baselocalization.loginScreen.loginScreenValidationError,
                baselocalization.loginScreen.loginScreenEmailNotFound,
            );
        } else if (userExist && user.password !== values.password) {
            Alert.alert(
                baselocalization.loginScreen.loginScreenValidationError,
                baselocalization.loginScreen.loginScreenIncorrectPassword,
            );
        } else if (userExist && user.password === values.password) {
            navigation.navigate('HomeScreen');
            dispatch(login_request(user));
        } else {
            Alert.alert(baselocalization.loginScreen.loginScreenCheckEmailAndPassword);
        }
    };

    return (
        <SafeAreaView>
            <ScrollView
                keyboardDismissMode="interactive"
                showsVerticalScrollIndicator={false}
                keyboardShouldPersistTaps="handled"
                style={styles.container}>
                <StatusBar barStyle="dark-content" backgroundColor={theme.colors.white} />
                <KeyboardAvoidingView style={styles.subContainer}>
                    <View style={styles.imageContainer}>
                        <Image
                            source={images.homescreen_mcdIcon}
                            style={styles.image}
                        />
                    </View>
                    <View style={styles.viewHeaderStyle}>
                        <Text style={styles.textHeaderStyle}>
                            {baselocalization.loginScreen.loginScreenWelcome}
                        </Text>
                        <Text style={styles.textHeaderStyle}>
                            {baselocalization.loginScreen.loginMsg}
                        </Text>
                    </View>
                    <View>
                        <SafeAreaView>
                            <View>
                                <Formik
                                    initialValues={initialValues}
                                    onSubmit={onSubmit}
                                    validationSchema={ValidationSchema().getLogInAccountValidationSchema()}
                                    //@ts-ignore
                                    innerRef={formikRef}>
                                    {({
                                        values,
                                        handleChange,
                                        errors,
                                        setFieldTouched,
                                        touched,
                                        handleSubmit,
                                    }) => (
                                        <View>
                                            <CustomTextInput
                                                testID='LoginScreen_EmailInputField'
                                                value={values.email}
                                                errorText={errors.email}
                                                onChangeText={handleChange('email')}
                                                onBlur={() => setFieldTouched('email')}
                                                isError={touched.email && errors.email ? true : false}
                                                labelText={baselocalization.loginScreen.loginScreenEmail}
                                                lablePlaceHolder={baselocalization.loginScreen.loginScreenEmailPlaceHolder}
                                            />
                                            <CustomTextInput
                                                testID='LoginScreen_PasswordInputField'
                                                value={values.password}
                                                errorText={errors.password}
                                                onChangeText={handleChange('password')}
                                                onBlur={() => setFieldTouched('password')}
                                                isError={
                                                    touched.password && errors.password ? true : false
                                                }
                                                labelText={baselocalization.loginScreen.loginScreenPassword}
                                                lablePlaceHolder={baselocalization.loginScreen.loginScreenPasswordPlaceHolder}
                                                secureTextEntry={true}
                                                isPassword={true}
                                            />
                                            <View style={styles.button}>
                                                <CustomButton testID='LoginScreen_NextButton' text={baselocalization.onboardingScreen.onboardingScreenButtonText}
                                                    disabled={false}
                                                    buttonStylesProps={{
                                                        backgroundColor: theme.colors.primary,
                                                        borderColor: theme.colors.primary,
                                                        textColor: theme.colors.white,
                                                        width: wp('80%'),
                                                        height: hp('5%'),
                                                        fontSize: undefined
                                                    }}
                                                    onPress={handleSubmit} />
                                            </View>
                                        </View>
                                    )}
                                </Formik>
                                <View style={styles.accountExistViewStyle}>
                                    <Text style={styles.accoutExistStyle}>
                                        {baselocalization.loginScreen.loginScreenNoAccountMsg}
                                    </Text>
                                    <TouchableOpacity
                                        onPress={() => {
                                            navigation.navigate('SignupScreen');
                                        }}>
                                        <Text style={styles.accoutExistBtnStyle}>
                                            {baselocalization.loginScreen.loginScreenSignUp}
                                        </Text>
                                    </TouchableOpacity>
                                </View>
                            </View>
                        </SafeAreaView>
                    </View>
                </KeyboardAvoidingView>
            </ScrollView>
        </SafeAreaView>
    );
};

export default LoginScreen;
