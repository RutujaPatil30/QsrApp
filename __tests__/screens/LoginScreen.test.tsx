import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import LoginScreen from '../../src/screens/LoginScreen/LoginScreen';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

const store = mockStore({
    reducer: {
        users: [
            { email: 'test@example.com', password: 'password@123' },
        ],
    },
});

describe('LoginScreen', () => {
    it('renders correctly', () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <LoginScreen />
            </Provider>
        );

        expect(getByTestId('LoginScreen_EmailInputField')).toBeTruthy();
        expect(getByTestId('LoginScreen_PasswordInputField')).toBeTruthy();
        expect(getByTestId('LoginScreen_NextButton')).toBeTruthy();
    });

    it('submits the form with correct credentials', () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <LoginScreen />
            </Provider>
        );

        fireEvent.changeText(getByTestId('LoginScreen_EmailInputField'), 'test@example.com');
        fireEvent.changeText(getByTestId('LoginScreen_PasswordInputField'), 'password123');
        fireEvent.press(getByTestId('LoginScreen_NextButton'));

    })

})

describe('LoginScreen Snapshot Test', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <LoginScreen />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
  });