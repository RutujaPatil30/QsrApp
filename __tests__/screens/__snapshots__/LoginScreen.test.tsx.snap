// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Snapshot Test renders correctly 1`] = `
<RCTScrollView
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  showsVerticalScrollIndicator={false}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": 750,
    }
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={
        {
          "alignSelf": "center",
          "width": 675,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <RCTImageView
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          internal_analyticTag={null}
          resizeMode="contain"
          source={
            [
              {
                "height": undefined,
                "uri": undefined,
                "width": undefined,
              },
            ]
          }
          style={
            {
              "height": 112.5,
              "overflow": "hidden",
              "resizeMode": "contain",
              "width": 225,
            }
          }
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "paddingVertical": 19,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000000",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Welcome To McDonald's
        </Text>
        <Text
          style={
            {
              "color": "#000000",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Login to continue
        </Text>
      </View>
      <View>
        <RCTSafeAreaView>
          <View>
            <View>
              <View
                style={
                  {
                    "flex": 1,
                    "padding": 13.5,
                  }
                }
                testID="LoginScreen_EmailInputField"
              >
                <Text
                  style={
                    {
                      "color": "#000000",
                      "fontSize": 16,
                    }
                  }
                >
                  Email
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#000000",
                      "flexDirection": "row",
                    }
                  }
                >
                  <TextInput
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Enter your email"
                    placeholderTextColor="#808080"
                    secureTextEntry={false}
                    style={
                      {
                        "backgroundColor": "#ffffff",
                        "color": "#000000",
                        "flex": 1,
                        "fontSize": 16,
                      }
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  {
                    "flex": 1,
                    "padding": 13.5,
                  }
                }
                testID="LoginScreen_PasswordInputField"
              >
                <Text
                  style={
                    {
                      "color": "#000000",
                      "fontSize": 16,
                    }
                  }
                >
                  Password
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#000000",
                      "flexDirection": "row",
                    }
                  }
                >
                  <TextInput
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="Enter your password"
                    placeholderTextColor="#808080"
                    secureTextEntry={true}
                    style={
                      {
                        "backgroundColor": "#ffffff",
                        "color": "#000000",
                        "flex": 1,
                        "fontSize": 16,
                      }
                    }
                    value=""
                  />
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                        "padding": 6.5,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#808080",
                            "fontSize": 24,
                          },
                          undefined,
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      ó°ˆˆ
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "flex-end",
                    "marginVertical": 19,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#da291c",
                      "borderColor": "#da291c",
                      "borderRadius": 53.5,
                      "borderWidth": 1,
                      "height": 66.5,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 600,
                    }
                  }
                  testID="LoginScreen_NextButton"
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000000",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        },
                        {
                          "color": "#ffffff",
                        },
                      ]
                    }
                  >
                    Pesan Sekarang
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 16,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#000000",
                    "fontSize": 16,
                  }
                }
              >
                Don't have an account?
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#da291c",
                      "fontSize": 16,
                    }
                  }
                >
                   Sign Up
                </Text>
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
